packages/itwin/grouping-mapping-widget/src/test/QueryBuilder.test.ts

```diff
@@ -2,178 +2,67 @@
-    it("should return empty string, when query unions are empty", () => testBuildQueryString({unions: []}, ""));
-    it("should return empty string, when query classes are empty", () => testBuildQueryString({unions: [{classes: []}]}, ""));
-
-    it("should return query string with ROUND, when property is float", () => {
-      const baseClass: QueryClass = {
-        className: "BaseClassName",
-        isAspect: false,
-        isRelational: true,
-        properties: [],
-      };
-
-      const joinClass: QueryClass = {
-        className: "JoinClassName",
-        isAspect: false,
-        isRelational: true,
-        properties: [
-          {
-            name: "propName1",
-            needsQuote: false,
-            isCategory: false,
-            value: 3.14159,
-          },
-        ],
-      };
-
-      const query: Query = {
-        unions: [
-          {classes: [baseClass, joinClass]},
-        ],
-      };
+  let sut: QueryBuilder;
+  let dataProvider: StubbedType<PresentationPropertyDataProvider>;
 
-      testBuildQueryString(query, "SELECT BaseClassName.ECInstanceId FROM BaseClassName JOIN JoinClassName ON ROUND(JoinClassName.propName1, 4)=3.1416");
-    });
+  beforeEach(() => {
+    dataProvider = MockFactory.create(PresentationPropertyDataProvider);
+    MockFactory.stubProperty(dataProvider, "getContentDescriptor", () => () => true);
 
-    it("should return query string with propery value in quotes, when property needs quotes", () => {
-      const baseClass: QueryClass = {
-        className: "BaseClassName",
-        isAspect: false,
-        isRelational: true,
-        properties: [],
-      };
-
-      const joinClass: QueryClass = {
-        className: "JoinClassName",
-        isAspect: false,
-        isRelational: true,
-        properties: [
-          {
-            name: "propName1",
-            needsQuote: true,
-            isCategory: false,
-            value: "someName",
-          },
-        ],
-      };
-
-      const query: Query = {
-        unions: [
-          {classes: [baseClass, joinClass]},
-        ],
-      };
-
-      testBuildQueryString(query, "SELECT BaseClassName.ECInstanceId FROM BaseClassName JOIN JoinClassName ON JoinClassName.propName1='someName'");
-    });
-
-    it("should return a category query string, when property is category", () => {
-      const baseClass: QueryClass = {
-        className: "BaseClassName",
-        isAspect: false,
-        isRelational: true,
-        properties: [],
-      };
+    sut = new QueryBuilder(dataProvider);
+  });
 
-      const joinClass: QueryClass = {
-        className: "JoinClassName",
-        isAspect: false,
-        isRelational: true,
-        properties: [
-          {
-            name: "propName1",
-            needsQuote: false,
-            isCategory: true,
-            value: "propValue1",
-          },
-        ],
-      };
+  afterEach(() => {
+    dataProvider.getFieldByPropertyRecord.reset();
```

Comment:
It might be better to `restore()` the original function.

Reply:
Changed to `.restore()`

Reply:
Resolved